packages/@aws-cdk/aws-cloudwatch/lib/dashboard.ts

```diff
@@ -67,6 +68,13 @@ export class Dashboard extends Resource {
       physicalName: props.dashboardName,
     });
 
+    if (props.dashboardName && !props.dashboardName.match(/^[\w-]+$/)) {
```

Comment:
You must check that `dashboardName` is not an unresolved token: `Token.isUnresolved(dashboardName)`. If it is an unresolved token, you can't perform the validation. See [Tokens](https://docs.aws.amazon.com/cdk/latest/guide/tokens.html)

Reply:
Fixed, thanks 👍